client/src/featureform/register.py

```diff
@@ -736,15 +736,8 @@ def __init__(self, fn, registrar, provider, decorator_register_resources_method,
         self.fn = fn
         self.registrar = registrar
         self.provider = provider
-        # Binds the register_resources and name_variant methods to the subscriptable_fn
-        # using the descriptor protocol. This allows us to call the methods on the
-        # subscriptable_fn instance. **NOTE** the MethodType could also be used to bind
-        # the methods to the subscriptable_fn instance; however, the descriptor protocol
-        # produces the same result while also being compatible with `classmethod`
-        # and `staticmethod`.
-        self.register_resources = decorator_register_resources_method.__get__(self)
-        self.name_variant = decorator_name_variant_method.__get__(self)
-        pass
+        self.register_resources = decorator_register_resources_method
```

Comment:
Ironically this is originally how I thought we should do this, and the whole __get__ thing confused the hell out of me. I just assume you can use fns and methods however you want in python.

Reply:
I probably overcomplicated this based on a misguided bias from JS. If this were JS, I would never expect the value of `self` to be correct unless I bound it. When I searched for how to do this in Python, I happened upon the approach, which was touted as better than just a "raw" assignment or using `types.MethodType`. Lesson learned.